dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/SubmissionController.java

```diff
@@ -0,0 +1,231 @@
+            String errorMessage = "The share link is empty.";
+            log.error(errorMessage);
+            throw new RuntimeException(errorMessage);
+        }
+
+        // Create a DTO with the share link for better processing in the FE
+        ShareSubmissionLinkDTO shareSubmissionLinkDTO = new ShareSubmissionLinkDTO();
+        shareSubmissionLinkDTO.setShareLink(shareLink);
+
+        // Send share link in response
+        return ResponseEntity.ok().body(shareSubmissionLinkDTO);
+    }
+
+    @PreAuthorize("hasPermission(#wsoId, 'WORKSPACEITEM', 'WRITE')")
+    @RequestMapping(method = RequestMethod.POST, value = "setOwner")
+    public WorkspaceItemRest setOwner(@RequestParam(name = "shareToken", required = false)
+                                                           String shareToken, HttpServletRequest request)
+            throws SQLException, AuthorizeException {
+
+        Context context = ContextUtil.obtainContext(request);
+        // Check the context is not null
+        this.validateContext(context);
+
+        // Get workspace by share token
+        List<WorkspaceItem> wsiList = workspaceItemService.findByShareToken(context, shareToken);
+        // Check the wsi does exist
+        if (CollectionUtils.isEmpty(wsiList)) {
+            String errorMessage = "The workspace item with share token:" + shareToken + " does not exist.";
+            log.error(errorMessage);
+            throw new BadRequestException(errorMessage);
+        }
+
+        // Get the first workspace item - the only one
+        WorkspaceItem wsi = wsiList.get(0);
+        // Check the wsi does exist
+        validateWorkspaceItem(wsi, null, shareToken);
+
+        if (!authorizeService.authorizeActionBoolean(context, wsi.getItem(), Constants.READ)) {
+            String errorMessage = "The current user does not have rights to view the WorkflowItem";
+            log.error(errorMessage);
+            throw new AccessDeniedException(errorMessage);
+        }
+
+        // Set the owner of the workspace item to the current user
+        EPerson currentUser = context.getCurrentUser();
+        // If the current user is null, throw an exception
+        if (currentUser == null) {
+            String errorMessage = "The current user is not valid, it cannot be null.";
+            log.error(errorMessage);
+            throw new BadRequestException(errorMessage);
+        }
+
+        wsi.getItem().setSubmitter(currentUser);
+        workspaceItemService.update(context, wsi);
+        WorkspaceItemRest wsiRest = converter.toRest(wsi, utils.obtainProjection());
+
+        // Without commit the changes are not persisted into the database
+        context.commit();
+        return wsiRest;
+    }
+
+
+
+    private static String generateShareToken() {
+        // UUID generates a 36-char string with hyphens, so we can strip them to get a 32-char string
+        return UUID.randomUUID().toString().replace("-", "").substring(0, 32);
```

Comment:
why is - a problem?

Reply:
It’s not a problem at all. The token just looks better without the -, that’s all.